File: structural/dart/bridge_structured/interfaces.dart
abstract class DrawingAPI {
  void drawCircle(double x, double y, double radius);
  void drawRectangle(double x, double y, double width, double height);
}

abstract class Device {
  void turnOn();
  void turnOff();
  void setVolume(int volume);
  int getVolume();
}

File: structural/typescript/bridge_structured/interfaces.ts
export interface DrawingAPI {
  drawCircle(x: number, y: number, radius: number): void;
  drawRectangle(x: number, y: number, width: number, height: number): void;
}

export interface Device {
  turnOn(): void;
  turnOff(): void;
  setVolume(volume: number): void;
  getVolume(): number;
}
